core/data_storage.py

```
@@ -0,0 +1,614 @@
+    
+    @contextmanager
+    def get_connection(self):
+        """Context manager for database connections."""
+        conn = sqlite3.connect(self.db_path)
+        conn.row_factory = sqlite3.Row  # Enable dict-like access
+        try:
+            yield conn
+            conn.commit()
+        except Exception as e:
+            conn.rollback()
+            raise e
```

Ruff (0.13.3)

Use `raise` without specifying exception name

Remove exception name

(TRY201)